File: src/main/java/quizapi/repository/CategoryRepository.java
package quizapi.repository;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import quizapi.model.Category;

import java.util.List;
import java.util.Optional;

@Repository
public interface CategoryRepository extends CrudRepository<Category, Long> {
    Optional<Category> findByName(String name);

    @Query("SELECT name FROM Category")
    List<String> findCategoriesNames();
}

File: src/main/kotlin/com/shkila/quizapi/repository/CategoryRepository.kt
package com.shkila.quizapi.repository

import com.shkila.quizapi.model.Category
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface CategoryRepository : CrudRepository<Category, Long> {
    fun findByName(name: String): Optional<Category>

    @Query("SELECT name FROM Category")
    fun findCategoriesNames(): List<String>
}